lighthouse-core/gather/connections/connection.js

```diff
@@ -19,13 +19,31 @@
 const EventEmitter = require('events').EventEmitter;
 const log = require('../../lib/log.js');
 
+/* eslint-disable no-unused-vars */
+
+/**
+ * @interface
+ */
+class ConnectionDelegate {
+  /**
+   * Evaluate an expression in the context of the current page.
+   * Returns a promise that resolves on the expression's value.
+   * @param {string} expression
+   * @return {!Promise<*>}
+   */
+  evaluateAsync(expression) { }
```

Comment:
this function already exists in `driver`, rather than implement this in each connection type thoughts on just using the userAgent method everywhere for now?